File: app/src/main/java/com/star_zero/example/androidmvvm/di/AppComponent.java
package com.star_zero.example.androidmvvm.di;

import com.star_zero.example.androidmvvm.presentation.add_edit_task.AddEditTaskActivity;
import com.star_zero.example.androidmvvm.presentation.tasks.TasksActivity;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = {AppModule.class})
@Singleton
public interface AppComponent {

    void inject(TasksActivity activity);

    void inject(AddEditTaskActivity activity);
}

File: app/src/main/kotlin/com/star_zero/example/androidmvvm/di/AppComponent.kt
package com.star_zero.example.androidmvvm.di

import android.app.Application
import com.star_zero.example.androidmvvm.App
import dagger.BindsInstance
import dagger.Component
import javax.inject.Singleton



@Component(modules = arrayOf(
        AppModule::class,
        ActivityBuilderModule::class))
@Singleton
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

    fun inject(app: App)
}